inc/saiport.h

```diff
@@ -170,6 +170,18 @@ typedef enum _sai_port_fec_mode_t
 
     /** Enable FC-FEC - 10G, 25G, 40G, 50G ports */
     SAI_PORT_FEC_MODE_FC,
+
+    /** Enable RS-FEC - CL91/RS528 - 100G NRZ ports */
+    SAI_PORT_FEC_MODE_CL91_RS,
+
+    /** Enable RS-FEC - CL108/RS528 - 25G, 50G NRZ ports */
+    SAI_PORT_FEC_MODE_CL108_RS,
+
+    /** Enable RS-FEC - CL134/RS544 - 50G PAM4 ports */
+    SAI_PORT_FEC_MODE_CL134_RS,
+
+    /** Enable RS-FEC - CL119/RS544 - 100G, 200G, 400G PAM4 ports */
+    SAI_PORT_FEC_MODE_CL119_RS,
```

Comment:
Can we just use RS or FC in SAI header file.

In vendor code, fec types/clauses can be addressed based on the interface type.
Other wise there would be lot of error conditions check need to be added on port speed and its fec combination mismatches